### PR TITLE
Plumb errors through TransactionStatusService instead of panicking

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
I was reviewing https://github.com/anza-xyz/agave/pull/3026 post-merge and noticed we had a handful of `.expect()` and `.unwrap()` in `TransactionStatusService`. These do not allow for graceful teardown of the process, so try to be nicer to other threads by initiating a more graceful teardown

#### Summary of Changes
- Rework TransactionStatusService to raise/handle errors instead of `.expect()` and `.unwrap()`
- Set validator exit flag if fatal error encountered
- Add logs for service start/stop/error encountered

One additional item I'll call out is that the panic hook pretty quickly kills the entire process via this:
https://github.com/anza-xyz/agave/blob/7b05e29f4eb275bf884b18351b2d5ad13dc48bf0/metrics/src/metrics.rs#L538-L539

The sentiment behind that comment is still valid. However, enough threads respect the exit flag such that the process will still come down fairly quickly so I think setting the flag is ok in this scenario

### CODE REVIEW

**Review by bw-solana — APPROVED** [review]
LGTM.

I like the change to have separate receive/process at the top level too. This seems to better match the pattern we use elsewhere.

Would be good to wait for @fkouteib to review before merging since he spent more time here recently

**Review by fkouteib — APPROVED** [review]
LGTM. Thanks for cleaning this up.

**Pull request merged**
